lib/Doctrine/MongoDB/Aggregation/Stage/Match.php

```diff
@@ -0,0 +1,604 @@
+     * @param float $minDistance
+     * @return self
+     */
+    public function minDistance($minDistance)
+    {
+        $this->query->minDistance($minDistance);
+
+        return $this;
+    }
+
+    /**
+     * Specify $mod criteria for the current field.
+     *
+     * @see Expr::mod()
+     * @see http://docs.mongodb.org/manual/reference/operator/mod/
+     * @param float|integer $divisor
+     * @param float|integer $remainder
+     * @return self
+     */
+    public function mod($divisor, $remainder = 0)
+    {
+        $this->query->mod($divisor, $remainder);
+
+        return $this;
+    }
+
+    /**
+     * Negates an expression for the current field.
+     *
+     * You can create a new expression using the {@link Builder::expr()} method.
+     *
+     * @see Expr::not()
+     * @see http://docs.mongodb.org/manual/reference/operator/not/
+     * @param array|Expr $expression
+     * @return self
+     */
+    public function not($expression)
+    {
+        $this->query->not($expression);
+
+        return $this;
+    }
+
+    /**
+     * Specify $ne criteria for the current field.
+     *
+     * @see Expr::notEqual()
+     * @see http://docs.mongodb.org/manual/reference/operator/ne/
+     * @param mixed $value
+     * @return self
+     */
+    public function notEqual($value)
+    {
+        $this->query->notEqual($value);
+
+        return $this;
+    }
+
+    /**
+     * Specify $nin criteria for the current field.
+     *
+     * @see Expr::notIn()
+     * @see http://docs.mongodb.org/manual/reference/operator/nin/
+     * @param array $values
+     * @return self
+     */
+    public function notIn(array $values)
+    {
+        $this->query->notIn($values);
+
+        return $this;
+    }
+
+    /**
+     * Specify $gte and $lt criteria for the current field.
+     *
+     * This method is shorthand for specifying $gte criteria on the lower bound
+     * and $lt criteria on the upper bound. The upper bound is not inclusive.
+     *
+     * @see Expr::range()
+     * @param mixed $start
+     * @param mixed $end
+     * @return self
+     */
+    public function range($start, $end)
+    {
+        $this->query->range($start, $end);
+
+        return $this;
+    }
+
+    /**
+     * Specify $size criteria for the current field.
+     *
+     * @see Expr::size()
+     * @see http://docs.mongodb.org/manual/reference/operator/size/
+     * @param integer $size
+     * @return self
+     */
+    public function size($size)
+    {
+        $this->query->size((integer) $size);
+
+        return $this;
+    }
+
+    /**
+     * Specify $text criteria for the current field.
+     *
+     * The $language option may be set with {@link Builder::language()}.
+     *
+     * @see Expr::text()
+     * @see http://docs.mongodb.org/master/reference/operator/query/text/
+     * @param string $search
+     * @return self
+     */
+    public function text($search)
+    {
+        $this->query->text($search);
```

Comment:
Yep, that was the thought. I would, however, point this out in the docblock, just like with ```$geoNear``` being the first stage in the pipeline.